tests/arp/test_wr_arp.py

```diff
@@ -0,0 +1,173 @@
+        @param ptfhost: AnsibleHost object of PTF
+        '''
+        mgFacts = duthost.minigraph_facts(host=duthost.hostname)['ansible_facts']
+        vxlanConfigData = {
+            'minigraph_port_indices': mgFacts['minigraph_port_indices'],
+            'minigraph_portchannel_interfaces': mgFacts['minigraph_portchannel_interfaces'],
+            'minigraph_portchannels': mgFacts['minigraph_portchannels'],
+            'minigraph_lo_interfaces': mgFacts['minigraph_lo_interfaces'],
+            'minigraph_vlans': mgFacts['minigraph_vlans'],
+            'minigraph_vlan_interfaces': mgFacts['minigraph_vlan_interfaces'],
+            'dut_mac': duthost.setup()['ansible_facts']['ansible_Ethernet0']['macaddress']
+        }
+        with open(VXLAN_CONFIG_FILE, 'w') as file:
+            file.write(json.dumps(vxlanConfigData, indent=4))
+
+        logger.info('Copying ferret config file to {0}'.format(ptfhost.hostname))
+        ptfhost.copy(src=VXLAN_CONFIG_FILE, dest='/tmp/')
+
+    @pytest.fixture(scope='class', autouse=True)
+    def setupFerret(self, duthost, ptfhost):
+        '''
+        Sets Ferret service on PTF host. This class-scope fixture runs once before test start
+
+        @param duthost: AnsibleHost object of DUT
+        @param ptfhost: AnsibleHost object of PTF
+        '''
+        ptfhost.copy(src="scripts/ferret.py", dest="/opt")
+
+        result = duthost.shell(
+            cmd='''ip route show proto zebra type unicast |
+            sed -e '/default/d' -ne '/0\//p' |
+            head -n 1 |
+            sed -ne 's/0\/.*$/1/p'
+            '''
+        )
+        assert len(result['stderr_lines']) == 0, 'Could not obtain DIP'
+
+        dip = result['stdout']
+        logger.info('VxLan Sender {0}'.format(dip))
+
+        ptfhost.host.options['variable_manager'].extra_vars.update({
+            'ferret_args': '-f /tmp/vxlan_decap.json -s {0}'.format(dip)
+        })
+
+        logger.info('Copying ferret config file to {0}'.format(ptfhost.hostname))
+        ptfhost.template(src='ferret.conf.j2', dest='/etc/supervisor/conf.d/ferret.conf')
+
+        logger.info('Generate pem and key files for ssl')
+        ptfhost.command(
+            cmd='''openssl req -new -x509 -keyout test.key -out test.pem -days 365 -nodes 
+            -subj "/C=10/ST=Test/L=Test/O=Test/OU=Test/CN=test.com"''',
+            chdir='/opt'
+        )
+
+        self.__prepareVxlanConfigData(duthost, ptfhost)
+
+        logger.info('Refreshing supervisor control with ferret configuration')
+        ptfhost.shell('supervisorctl reread && supervisorctl update')
+
+    @pytest.fixture(scope='class', autouse=True)
+    def copyPtfDirectory(self, ptfhost):
+        '''
+        Copys PTF directory to PTF host. This class-scope fixture runs once before test start
+
+        @param ptfhost: AnsibleHost object of PTF
+        '''
+        ptfhost.copy(src="ptftests", dest="/root")
+
+    @pytest.fixture(scope='class', autouse=True)
```

Comment:
This is cool - does autouse just automatically run these fixtures for this file or does it have some other behavior as well?

Reply:
they get run automatically. I decided to follow different pattern for this conversion so it would help whoever would like to add new test.

Reply:
It was *very* helpful to me, so I'd call that a success. 😛